packs/BP/scripts/network_ipc.ts

```diff
@@ -106,19 +106,24 @@ export function generateListener(payload: ipc.SerializableValue): null {
   const location = deserializeDimensionLocation(data.a);
   const type = data.b;
   const amount = data.c;
+  const useReserveStorage = data.d;
+  const consumeExisting = data.e;
 
   const block = location.dimension.getBlock(location);
   if (!block) return null;
 
-  const fullAmount = amount + getMachineStorage(location, type);
-  if (!fullAmount) return null;
+  // If consume existing is true, do not add new power, instead consume the existing storage
+  const newAmount =
+    getMachineStorage(location, type) + (consumeExisting ? 0 : amount);
 
-  const storageType = InternalRegisteredStorageType.forceGetInternal(type);
+  // Set the new amount, as *send will take it directly out of storage
+  setMachineStorage(block, type, newAmount);
```

Comment:
nothing is added if `consumeExisting` is `true`, so we only need to call `setMachineStorage` if `consumeExisting` is `false`
```suggestion
  if (!consumeExisting) {
    setMachineStorage(block, type, newAmount);
  }
```